src/ng/directive/input.js

```diff
@@ -1126,7 +1128,11 @@ function baseInputType(scope, element, attr, ctrl, $sniffer, $browser) {
     });
   }
 
-  var timeout;
+  var timeout, oldVal;
+  var viewValueUpdated = false, msieInput = msie >= 10 && msie <= 11;
+  if (msieInput && attr.type in IE_INPUTS_WITH_CLEARING) {
```

Comment:
Wouldn't that trigger a potentially unnecessary DOM access though? That is likely more expensive than this check I think. I originally had it just accessing the value, I'd be fine changing it back.